### PR TITLE
feat: init proposal detail page new design

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Initialize new proposal detail UI and present type+topic information including their accurate description.

# Changes

- new `<ProposalModern />` cmp for the detail page behind feature flag
- existing legacy code move to `<ProposalLegacy />`
- remove unused `<InfoContextKey />` cmp
- replace and remove `<TwoColumns />` cmp with style class of `gix-components`
- copy all type and topic description (i18n labels) of the dashboard
- introduce new `<KeyValuePairInfo />`. like `KeyValuePair` plus an info button to display more information (click only on button as discussed with Mischa)
- update `<Collapsible />` to make "toggle" (display / hide more info) accessible from outside of the cmp

# Screenshots

Capture d’écran 2022-08-23 à 15 44 37

Enregistrement de l’écran 2022-08-23 à 15 44 44

### CODE REVIEW

File: frontend/src/lib/components/proposal-detail/ProposalSystemInfoSection.svelte [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  import type { ProposalInfo } from "@dfinity/nns";
+  import { mapProposalInfo } from "../../utils/proposals.utils";
+  import KeyValuePairInfo from "../ui/KeyValuePairInfo.svelte";
+  import { i18n } from "../../stores/i18n";
+  import { sanitizeHTML } from "../../utils/html.utils";
+
+  export let proposalInfo: ProposalInfo;
+
+  let type: string | undefined;
+  let type_description: string | undefined;
```
Comment (mstrasinskis) [inline frontend/src/lib/components/proposal-detail/ProposalSystemInfoSection.svelte:11]: nitpick: Probably `camelCase` would look better in the code.

File: frontend/src/lib/components/ui/KeyValuePairInfo.svelte [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  export let testId: string | undefined = undefined;
+
+  let collapsibleRef: Collapsible | undefined = undefined;
+</script>
+
+<Collapsible
+  expandButton={false}
+  externalToggle={true}
+  bind:this={collapsibleRef}
```
Comment (mstrasinskis) [inline frontend/src/lib/components/ui/KeyValuePairInfo.svelte:14]: nth
```
let toggleContent: () => void;
...
bind:toggleContent={toggleContent}
```

**Review by mstrasinskis — APPROVED** [review]
LGTM, a couple of minor comments

File: frontend/src/lib/components/proposal-detail/ProposalSystemInfoSection.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/proposal-detail/ProposalSystemInfoSection.svelte]: Haha that was my first implementation and then I was sceptical about the outcome 😉. You are right, done.

File: frontend/src/lib/components/ui/KeyValuePairInfo.svelte (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/ui/KeyValuePairInfo.svelte]: Wow ❤️ it!

Done, thx!

**Pull request merged**